### ISSUE TITLE
Confused with CLIP output embedding: "laion/CLIP-ViT-L-14-laion2B-s32B-b82K"

### ISSUE THREAD

**Issue by 25icecreamflavors** [mlfoundations/open_clip#189]
I use this model in tensorflow (from huggingface): `TFCLIPVisionModel.from_pretrained("laion/CLIP-ViT-L-14-laion2B-s32B-b82K", from_pt=True)` and also I use the same model in pytorch `open_clip.create_model_and_transforms('ViT-L-14', 'laion2b_s32b_b82k')`. However, it appeared that the first one gives output as a (1024, 1) and the last one gives (768, 1). Why is it so?

**Comment by rwightman** [comment]
@25icecreamflavors I'm not familiar with the HF TF CLIP model impl, but you can see from the L/14 config, the common embedding dim for the output is 768, the vision tower itself has a wider 1024 output. Perhaps you are looking at the vision tower embedding on the TF model and not the projected?

``` json

{
    "embed_dim": 768,
    "vision_cfg": {
        "image_size": 224,
        "layers": 24,
        "width": 1024,
        "patch_size": 14
    },
    "text_cfg": {
        "context_length": 77,
        "vocab_size": 49408,
        "width": 768,
        "heads": 12,
        "layers": 12
    }
}
```

https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/model_configs/ViT-L-14.json

**Comment by 25icecreamflavors** [comment]
Yes, I think so. How can I get (1024, 1) embedding from the image on pytorch?

**Comment by rwightman** [comment]
@25icecreamflavors you can set model.visual.proj = None, BUT this isn't what most people want, while they're likely perfectly fine features, the major feature of CLIP is that the final projected embeddings of each tower are in the same embedding space, allowing image -> text, text -> image retreival, mapping of texts and images to a space where you can measure distance, etc ... the 1024 features won't be usable in that fashion with the text

**Comment by 25icecreamflavors** [comment]
Thank you, that's what I needed, because I worked only with images.
